test/e2e/memcached_test.go

```diff
@@ -574,3 +581,88 @@ func MemcachedClusterTest(t *testing.T) {
 		t.Fatalf("In-cluster test failed: %v\nCommand Output:\n%s", err, string(cmdOut))
 	}
 }
+
+func memcachedMetricsTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	operatorName := "memcached-operator"
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return err
+	}
+
+	// Make sure metrics Service exists
+	s := v1.Service{}
+	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: operatorName, Namespace: namespace}, &s)
+	if err != nil {
+		return fmt.Errorf("metrics Service does not exist: (%v)", err)
+	}
+
+	// Get operator pod
+	pods := v1.PodList{}
+	opts := client.ListOptions{Namespace: namespace}
+	if err := opts.SetLabelSelector("name=memcached-operator"); err != nil {
+		return fmt.Errorf("failed to set list label selector: (%v)", err)
+	}
+	if err := opts.SetFieldSelector("status.phase=Running"); err != nil {
+		return fmt.Errorf("failed to set list field selector: (%v)", err)
+	}
+	err = f.Client.List(context.TODO(), &opts, &pods)
+	if err != nil {
+		return fmt.Errorf("failed to get pods: (%v)", err)
+	}
+	podName := ""
+	if len(pods.Items) > 1 {
+		// If we got more than one pod, get leader pod name.
+		leader, err := verifyLeader(t, namespace, f)
+		if err != nil {
+			return err
+		}
+		podName = leader.Name
+	}
+	if podName == "" {
+		podName = pods.Items[0].Name
+	}
+
+	// Get metrics data
+	request := proxyViaPod(f.KubeClient, namespace, podName, "8383", "/metrics")
+	response, err := request.DoRaw()
+	if err != nil {
+		return fmt.Errorf("failed to get response from metrics: %v", err)
+	}
+
+	// Make sure metrics are present
+	body := string(response[:])
+	if body == "" {
```

Comment:
```suggestion
	if len(response) == 0 {
```
and remove the line above.